api/core.py

```diff
@@ -416,7 +430,11 @@ def _archivestream(self, ticket):
         stream = cStringIO.StringIO()
         with tarfile.open(mode='w|', fileobj=stream) as archive:
             for filepath, arcpath, _ in ticket['target']:
-                yield archive.gettarinfo(filepath, arcpath).tobuf()
+                try:
+                    yield archive.gettarinfo(filepath, arcpath).tobuf()
+                except Exception as e:
```

Comment:
I'm mainly trying to make sure that a web user (who won't see the log) won't be in a position to get a tarball that has fewer files than desired. I should clarify that by "incomplete tarballs" I didn't mean invalid ones, just ones with fewer files.

I can see how, if some files are corrupted, you might want to be able to retrieve any nondamaged files, but that strikes me as more of a disaster-recovery situation, and not a default desired behaviour.

I'm going to claim that we should abort entirely in that scenario, because I really don't want the user walking off thinking they have a complete set of data. I would want an entirely separate set of interactions that could check the availability of data, give a list of specific problems discovered, and selectively download the recoverable data.